### ISSUE TITLE
Provide debug for PostgeSQL procedures

### ISSUE THREAD

**Issue by ruspl-afed** [dbeaver/dbeaver#2556]
Debug functionality should be available for PL/pgSQL procedures

**Comment by serge-rider** [comment]
Depends on #1856

**Comment by zhangyc145** [comment]
Did you implement the debugger of pl/pgsql base on the plugin_debuggger in postgresql?Or some way else?

**Comment by serge-rider** [comment]
@zhangyc145 Yes, we going to use plugin_debugger.

**Comment by cvincentraj** [comment]
Hi, Can you help me on how to use pl/pgsql **Debugger** in **DBeaver** if this feature is already available?

Right now I am using DBeaver Version 4.2.5

Please guide me the steps, incase any installation required.


Thanks - Vin¢

**Comment by serge-rider** [comment]
It is still in development.

**Comment by ruspl-afed** [comment]
There are still several usability issues to be covered:

1. Allow to enter script for "Local" debug during initial launch creation
2. Separate launch shortcuts for "Local" and "Global" debug
3. Allow to enter PID for "Global" debug during initial launch creation
4. Support "set value" for variables during debug
5. Allow to retarget saved launch configuration to another procedure

We can either continue here or create separate tickets

**Comment by serge-rider** [comment]
Great job 👍 
Closing. We'll make separate tickets for improvements.

**Comment by serge-rider** [comment]
In CE version debugger can be installed as extension: https://marketplace.eclipse.org/content/dbeaver-sql-debugger (because it is in beta state).
In future we going to include it in the main build.

**Comment by sgrinko** [comment]
On the server itself, I already installed the extension `CREATE EXTENSION pldbgapi SCHEMA public;` and connected it in the `postgresql.conf` - `shared_preload_libraries = 'plugin_debugger'`

Please write the steps to click where to start debugging the Plpgsql functions...
I'm using the version - Community Edition 5.0.2

**Comment by ruspl-afed** [comment]
Thank you for your interest @sgrinko , your experience is very valuable for us.

There is a usability issue addressed with #3255 : one need to activate Eclipse Debug functionality first to see the "Debug" menu item. 
The primary scenario is:
1) open the procedure from the "Databases" view
2) switch to "Sources" tab
3) enable the "Omit procedure header" to sync line numbers #3257 
4) Use "Debug As" -> "PL/pgSQL Procedure" context menu item to configure and launch the debug 

I suggest to continue this discussions using new ticket(s) you can create for particular issue to simplify tracking for us.
Looking forward for your suggestions regarding enchancments. Also, please do not hesitate to open PRs
